pymc_marketing/mmm/fourier.py

```diff
@@ -422,25 +490,58 @@ def sample_prior(self, coords: dict | None = None, **kwargs) -> xr.Dataset:
         coords[self.prefix] = self.nodes
         return self.prior.sample_prior(coords=coords, name=self.variable_name, **kwargs)
 
-    def sample_curve(self, parameters: az.InferenceData | xr.Dataset) -> xr.DataArray:
-        """Create full period of the fourier seasonality.
+    def sample_curve(
+        self,
+        parameters: az.InferenceData | xr.Dataset,
+        use_dates: bool = False,
+        start_date: datetime.datetime | None = None,
+    ) -> xr.DataArray:
+        """Create full period of the Fourier seasonality.
 
         Parameters
         ----------
         parameters : az.InferenceData | xr.Dataset
-            Inference data or dataset containing the fourier parameters.
+            Inference data or dataset containing the Fourier parameters.
             Can be posterior or prior.
+        use_dates : bool, optional
+            If True, use datetime coordinates for the x-axis. Defaults to False.
+        start_date : datetime.datetime, optional
+            Starting date for the Fourier curve. If not provided and use_dates is True,
+            it will be derived from the current year or month. Defaults to None.
 
         Returns
         -------
         xr.DataArray
-            Full period of the fourier seasonality.
+            Full period of the Fourier seasonality.
 
         """
         full_period = np.arange(self.days_in_period + 1)
-        coords = {
-            "day": full_period,
-        }
+
+        coords = {}
+        if use_dates:
+            if start_date is None:
+                today = datetime.datetime.now()
+                if isinstance(self, YearlyFourier):
+                    start_date = datetime.datetime(year=today.year, month=1, day=1)
+                elif isinstance(self, MonthlyFourier):
+                    start_date = datetime.datetime(
+                        year=today.year, month=today.month, day=1
+                    )
+                else:
+                    raise ValueError("Unknown Fourier type for deriving start_date")
+
+            date_range = pd.date_range(
+                start=start_date,
```

Comment:
pd.date_range should handle the conversions above, does it not?

Reply:
elaborate?

Reply:
no need to convert to datetime to put into this function as it is pretty flexible itself: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.date_range.html

But getting start date might require that conversion. I need to look at it in its entirety

Reply:
will have a look today and some tests too! low brain cells yesterday!

Reply:
Please have a look now @wd60622?

- I have removed all the pd.DateTime conversions where it was not needed as mentioned by you

> pd.date_range should handle the conversions above.


- and I added more tests in `test_fourier.py`

> Could you write a test for the sample plot workflow using dates